src/ipc/handler.c

```diff
@@ -800,6 +802,153 @@ static int ipc_glb_gdb_debug(uint32_t header)
 
 }
 
+#if CONFIG_PROBE
+static inline int ipc_probe_init(uint32_t header)
+{
+	struct sof_ipc_probe_dma_set_params *params = _ipc->comp_data;
+	int dma_set_up = (params->hdr.size - sizeof(*params)) /
+			 sizeof(struct probe_dma);
+
+	trace_ipc("ipc_probe_init()");
+
+	return probe_init(dma_set_up ? params->probe_dma : NULL);
+}
+
+static inline int ipc_probe_deinit(uint32_t header)
+{
+	trace_ipc("ipc_probe_deinit()");
+
+	return probe_deinit();
+}
+
+static inline int ipc_probe_dma_set(uint32_t header)
+{
+	struct sof_ipc_probe_dma_set_params *params = _ipc->comp_data;
+	uint32_t dmas_count = (params->hdr.size - sizeof(*params)) /
+			      sizeof(struct probe_dma);
```

Comment:
why not make the DMAs count explicit in the IPC structure?

Reply:
Interface is to align with cAVS one., that for starters.
Secondly, payload is to be platform-agnostic. Many GETters can be described by variable size - host does not know upfront what's the count of injection dmas or probe points.

If you say - then host should store such numbers - then we end up in a lot of rubbish code (when you account for all the getter variants).
If you say - dmas/ probe points count is platform specific - then you tie struct declaration with platform-FW definition. Apart from that, providing zero-ed payload (e.g. dma[6] with only one used yields you payload of [1] and [5] of zeros) is really a bad design as it requires additional checks on the host side.

That's why GETters in cAVS have data_offset_size field. SOF pays for the lack of getters in general. To resolve it in quick and elegant fashion, hdr::size calculation gets involved. I think Arthur and Adrian done nicely with thier approach.

Reply:
then use a macro or something that shows how to get the information in a repeatable/documented manner.

Reply:
@plbossart This is a very good idea. I'll do so within kernel and I urge @akloniex to do the same here.